core/include/io.h

```diff
@@ -273,4 +273,22 @@ static inline void io_clrsetbits8(vaddr_t addr, uint8_t clear_mask,
 	io_write8(addr, (io_read8(addr) & ~clear_mask) | set_mask);
 }
 
+#define readl_poll_timeout(addr, val, cond, delay_us, timeout_us) \
+({ \
+	uint64_t timeout = 0; \
+	uint64_t _delay_us = delay_us; \
+	bool flag = false; \
```

Comment:
Variables `timeout` and `flag` may conflict with variables with same label already defined in caller function. Could you add an underscore prefix to address that, as you did for `_delay_us` (`_timeout`, `_flag`)?

Reply:
```c
CHECK: Macro argument reuse '_addr' - possible side-effects?
```
checkpatch got such warning,`_val` and `_cond' have same warning.

Maybe i should change such argument  to addr_?